packages/core/src/sync-realtime/index.ts

```diff
@@ -233,10 +276,22 @@ export const createRealtimeSync = (
     for (const log of logs) {
       transactionHashes.add(log.transactionHash);
     }
-    for (const trace of callTraces) {
+    for (const trace of traces) {
       transactionHashes.add(trace.transactionHash);
     }
 
+    // TODO: handle factory address
+    transactions = transactions.filter((transaction) => {
+      let isMatched = transactionHashes.has(transaction.hash);
+      for (const filter of transactionFilters) {
+        if (isTransactionFilterMatched({ filter, block, transaction })) {
+          matchedFilters.add(filter);
+          isMatched = true;
+        }
+      }
+      return isMatched;
+    });
+
     transactions = transactions.filter((t) => transactionHashes.has(t.hash));
     transactionReceipts = transactionReceipts.filter((t) =>
```

Comment:
We should revisit this once we implement the `includeTransactionReceipt` feature